batdata/extractors/tIVT.py

```diff
@@ -0,0 +1,118 @@
+"""Extractor for .npy tIVT-format files"""
+from typing import Union, List, Iterator, Tuple
+
+import numpy as np
+import pandas as pd
+
+from batdata.extractors.base import BatteryDataExtractor
+from batdata.schemas import ChargingState
+from batdata.utils import drop_cycles
+from batdata.postprocess import add_steps, add_method, add_substeps
+from batdata.postprocess.cycle_stats import compute_capacity_energy
+
+from scipy.interpolate import interp1d
+from scipy.optimize import differential_evolution
+from scipy.signal import medfilt
+
+
+class tIVT_Extractor(BatteryDataExtractor):
+    """Parser for reading from .npy tIVT files
+
+    Expects the files to be in .npy format
+    """
+
+    def group(self, files: Union[str, List[str]], directories: List[str] = None,
+              context: dict = None) -> Iterator[Tuple[str, ...]]:
+        for file in files:
+            if file.lower().endswith('.npy'):
+                yield file
+
+    def generate_dataframe(self, file: str, file_number: int = 0, start_cycle: int = 0,
+                           start_time: float = 0) -> pd.DataFrame:
+        """Generate a DataFrame containing the data in this file
```

Comment:
You don't need to copy the docstring from a superclass, and should delete this for brevity.

Reply:
Will do! Sorry for the delay, I didn't realize something was waiting on me!